### PR TITLE
Fix ClassCastException while querying implicit nested arrays

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB

Fixes part of bug #13504 

```
CREATE TABLE doc.test ( 
"a" array(object as ( 
"b" array(object as ( 
"s" string 
)))));
insert into test (a) values ([{b=[{s='1'}, {s='2'}, {s='3'}]}]);
select a['b'] from test;

Before:
ClassCastException[class java.util.HashMap cannot be cast to class [Ljava.lang.Object; (java.util.HashMap and [Ljava.lang.Object; are in module java.base of loader 'bootstrap')]

After:
+----------------------------------------+
| a['b']                                 |
+----------------------------------------+
| [[{"s": "1"}, {"s": "2"}, {"s": "3"}]] |
+----------------------------------------+
```

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/expression/reference/doc/lucene/SourceParserTest.java [hunk 1]
```diff
@@ -213,4 +214,33 @@ public void test_null_object_next_not_sibling_column_has_same_name() {
...
+        // ex:
+        //   CREATE TABLE test (
+        //   "a" array(object as (
+        //   "b" array(object as (
+        //   "s" string
+        //   )))));
+        //   SELECT a['b'] from test; -- a['b'] is array(array(object))
+        ArrayType<?> type = new ArrayType<>(new ArrayType<>(ObjectType.builder().setInnerType("s", DataTypes.STRING).build()));
+        sourceParser.register(new ColumnIdent("_doc", List.of("a", "b")), type);
+        sourceParser.parse(new BytesArray(
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/expression/reference/doc/lucene/SourceParserTest.java:231]: Can you add assertions for the result?

File: docs/appendices/release-notes/unreleased.rst [hunk 2]
```diff
@@ -71,6 +71,9 @@ Fixes
 - Fixed an issue that caused the returned column names to be missing the
   subscripts when querying sub-columns of nested object arrays.
 
+- Fixed an issue that caused ``ClassCastException`` when accessing a sub-column
+  of a nested object array where the sub-column resolves to a nested array.
```
Comment (matriv) [inline docs/appendices/release-notes/unreleased.rst:75]: Maybe add an example here to make it 100% clear for users

**Review by matriv — APPROVED** [review]
LGTM, thank you! Left a comment about changelog

File: docs/appendices/release-notes/unreleased.rst (hunk 2 above)
Comment (jeeminso) [reply docs/appendices/release-notes/unreleased.rst:75]: Thank you, added an example.

**Pull request merged**